src/odemis/driver/test/keysight_test.py

```diff
@@ -107,15 +110,21 @@ def test_delay(self):
         """
         self.dev.period.value = 1e-4
         self.dev.delay.value = 1e-5
-        self.assertEqual(self.dev.delay.value, 1e-5)
+        self.assertAlmostEqual(self.dev.delay.value, 1e-5)
         self.dev.delay.value = 1e-6
-        self.assertEqual(self.dev.delay.value, 1e-6)
+        self.assertAlmostEqual(self.dev.delay.value, 1e-6)
+        self.dev.delay.value = -1e-6
+        self.assertAlmostEqual(self.dev.delay.value, -1e-6)
```

Comment:
if the delay time calculation depends on the sign of delay phase, maybe we can add test cases for that?

Reply:
What do you mean more precisely? The phase is just proportional to the delay (+ period). It's automatically computed by the device. So it's pretty simple. This part of the test case checks that it's possible to put a negative value. What would you check more?

Reply:
I am not sure what negative delay means. I understand the delay to be a positive value that adds up to delay the received signal by that much time. If the delay is negative, how is the timing of the signal interpreted ? I thought negative phase for e.g. -30 would be converted to positive value of 330. Then the delay will be based on 330. But I guess this is not the case.

My comment in the test case was based on the above interpretation. If the formulation of positive and negative phase delay differs, we can test the formulation.

Reply:
You're right, the delay in sec in converted into a phase. You are also right that a phase of -30° == +330°.
The device supports both. It accepts -360° to 360°. The driver also supports both "directions" for the delay. So we just pass it as-is: if the user talks about a negative delay, we convert to a negative phase. If the user talks about positive delay, we convert to positive phase.

Reply:
I've added an extra comment next to the VA to clarify that both negative and positive values are fine. The docstrings of both getPhase() and setPhase() already indicate that both signs are valid.